Tests/RequestInterceptorTests.swift

```diff
@@ -258,6 +319,35 @@ final class InterceptorTestCase: BaseTestCase {
         XCTAssertTrue(result.error is MockError)
     }
 
+    func testThatInterceptorCanAdaptRequestAsynchronously() {
+        // Given
+        let urlRequest = URLRequest(url: URL(string: "https://httpbin.org/get")!)
+        let session = Session()
+
+        let adapter = Adapter { urlRequest, _, completion in
+            DispatchQueue.main.async {
+                completion(.failure(MockError()))
+            }
+        }
+        let interceptor = Interceptor(adapters: [adapter])
+
+        var result: Result<URLRequest>!
+
+        let completesExpectation = expectation(description: "interceptor completes")
+
+        // When
+        interceptor.adapt(urlRequest, for: session) {
+            result = $0
+            completesExpectation.fulfill()
+        }
+
+        _ = waitForExpectations(timeout: 0.01)
```

Comment:
Please use our default timeout. It should be available as `timeout`. Also, there doesn't seem to be a need for the assignment here.